### PR TITLE
[EN DatetimeV2] Add parse for "once a week"

### PR DESCRIPTION
**Pull request by Grey0202**
- Add regex for "once a(n)".
- Add test

### CODE REVIEW

File: .NET/Microsoft.Recognizers.Definitions.Common/Chinese/DateTimeDefinitions.cs [hunk 1]
```diff
@@ -71,7 +71,7 @@ public static class DateTimeDefinitions
       public static readonly string SimpleCasesRegex = $@"((从)\s*)?(({YearRegex}|{DatePeriodYearInChineseRegex})\s*)?{MonthSuffixRegex}({DatePeriodDayRegexInChinese}|{DayRegex})\s*{DatePeriodTillRegex}\s*({DatePeriodDayRegexInChinese}|{DayRegex})((\s+|\s*,\s*){YearRegex})?";
       public static readonly string YearAndMonth = $@"({DatePeriodYearInChineseRegex}|{YearRegex})\s*{MonthRegex}";
       public static readonly string PureNumYearAndMonth = $@"({YearRegexInNumber}\s*[-\.\/]\s*{MonthNumRegex})|({MonthNumRegex}\s*\/\s*{YearRegexInNumber})";
-      public static readonly string OneWordPeriodRegex = $@"(((?<yearrel>(明|今|去)年)\s*)?({MonthRegex}|{HalfYearRegex})|({DatePeriodThisRegex}|{DatePeriodLastRegex}|{DatePeriodNextRegex})(?<halfTag>半)?\s*(周末|周|月|年)|周末|(今|明|去|前|后)年(\s*{HalfYearRegex})?)";
+      public static readonly string OneWordPeriodRegex = $@"(((?<yearrel>(明|今|去)年)\s*)?{MonthRegex}|({DatePeriodThisRegex}|{DatePeriodLastRegex}|{DatePeriodNextRegex})(?<halfTag>半)?\s*(周末|周|月|年)|周末|(今|明|去|前|后)年(\s*{HalfYearRegex})?)";
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Definitions.Common/Chinese/DateTimeDefinitions.cs]: Why remove HalfYearRegex?

File: Specs/DateTime/English/DateTimeModel.json [hunk 2]
```diff
@@ -14174,5 +14174,51 @@
         }
       }
     ]
+  },
+  {
+    "Input": "I'll leave once a week",
```
Comment (tellarin) [inline Specs/DateTime/English/DateTimeModel.json]: Can you change to something like "Let's meet once a week"?

File: Specs/DateTime/English/DateTimeModel.json [hunk 3]
```diff
@@ -14174,5 +14174,51 @@
...
+              "type": "set",
+              "value": "not resolved"
+            }
+          ]
+        }
+      }
+    ]
+  },
+  {
+    "Input": "I'll leave once a year",
```
Comment (tellarin) [inline Specs/DateTime/English/DateTimeModel.json]: Maybe "I go on vacation once a year"?

File: Specs/DateTime/English/SetParser.json [hunk 4]
```diff
@@ -619,5 +619,51 @@
         "Length": 7
       }
     ]
+  },
+  {
+    "Input": "I'll leave once a year",
+    "Context": {
+      "ReferenceDateTime": "2017-09-27T17:25:54.2909444+08:00"
```
Comment (tellarin) [inline Specs/DateTime/English/SetParser.json]: Can you use a regularly formed ReferenceDateTime, like in other specs?

File: .NET/Microsoft.Recognizers.Definitions.Common/Chinese/DateTimeDefinitions.cs (hunk 1 above)
Comment (Grey0202) [reply .NET/Microsoft.Recognizers.Definitions.Common/Chinese/DateTimeDefinitions.cs]: > Why remove HalfYearRegex?

It seems others changed the Yaml but forget to regenerate this file.

**Pull request merged**